mgmtfn/dockplugin/netDriver.go

```diff
@@ -371,9 +373,50 @@ func leave(w http.ResponseWriter, r *http.Request) {
 }
 
 func allocateNetwork(w http.ResponseWriter, r *http.Request) {
-	resp := api.AllocateNetworkResponse{}
 
-	content, err := json.Marshal(resp)
+	var (
+		content []byte
+		err     error
+		decoder = json.NewDecoder(r.Body)
+		anreq   = api.AllocateNetworkRequest{}
+	)
+
+	logEvent("allocateNetwork")
+
+	err = decoder.Decode(&anreq)
+	if err != nil {
+		httpError(w, "Could not read and parse the create network request", err)
+		return
+	}
+
+	log.Infof("AllocateNetworkRequest: %+v", anreq)
+
+	// If we are in docker-compose mode (swarm-mode):
+	//     - if tag is given, it indicates the epg or network
+	//     - else if subnet given create a contiv network
+	// In swarm-mode libnetwork api for allocateNetwork is called
+	// once in the cluster and createNetwork is called in every
+	// node where a container in the network is instantiated.
+	// We process only allocate network in this case.
+	//
+	if pluginMode == "docker-compose" {
```

Comment:
@rhim Today we have pluginMode in netplugin which indicates whether we are running in docker mode or kubernetes. PluginMode can have values "docker", "kubernetes" or "mesos". Since the new workflow is a special case of docker and it enables support for docker compose I chose the name 'docker-compose'. It is only valid in swarm-mode, but in future we could make it work in classic-swarm as well (libnetwork APIs are different). If you have any other name preference, I can change it

We should keep it in the same file as the changes specific to new workflow are only in a few of the APIs in this file